core/src/main/java/org/apache/gravitino/hook/ModelHookDispatcher.java

```diff
@@ -0,0 +1,176 @@
+          Owner.Type.USER);
+    }
+    return model;
+  }
+
+  @Override
+  public boolean deleteModel(NameIdentifier ident) {
+    return dispatcher.deleteModel(ident);
+  }
+
+  @Override
+  public int[] listModelVersions(NameIdentifier ident) throws NoSuchModelException {
+    return dispatcher.listModelVersions(ident);
+  }
+
+  @Override
+  public ModelVersion getModelVersion(NameIdentifier ident, int version)
+      throws NoSuchModelVersionException {
+    return dispatcher.getModelVersion(ident, version);
+  }
+
+  @Override
+  public ModelVersion getModelVersion(NameIdentifier ident, String alias)
+      throws NoSuchModelVersionException {
+    return dispatcher.getModelVersion(ident, alias);
+  }
+
+  @Override
+  public void linkModelVersion(
+      NameIdentifier ident,
+      String uri,
+      String[] aliases,
+      String comment,
+      Map<String, String> properties)
+      throws NoSuchModelException, ModelVersionAliasesAlreadyExistException {
+    dispatcher.linkModelVersion(ident, uri, aliases, comment, properties);
+  }
+
+  @Override
+  public boolean deleteModelVersion(NameIdentifier ident, int version) {
+    return dispatcher.deleteModelVersion(ident, version);
+  }
+
+  @Override
+  public boolean deleteModelVersion(NameIdentifier ident, String alias) {
+    return dispatcher.deleteModelVersion(ident, alias);
+  }
+
+  @Override
+  public boolean modelExists(NameIdentifier ident) {
+    return dispatcher.modelExists(ident);
+  }
+
+  @Override
+  public Model registerModel(
+      NameIdentifier ident,
+      String uri,
+      String[] aliases,
+      String comment,
+      Map<String, String> properties)
+      throws NoSuchSchemaException, ModelAlreadyExistsException,
+          ModelVersionAliasesAlreadyExistException {
+    // Check whether the current user exists or not
+    AuthorizationUtils.checkCurrentUser(
+        ident.namespace().level(0), PrincipalUtils.getCurrentUserName());
+
+    Model model = dispatcher.registerModel(ident, uri, aliases, comment, properties);
+
+    // Set the creator as owner of the model.
+    OwnerManager ownerManager = GravitinoEnv.getInstance().ownerManager();
+    if (ownerManager != null) {
+      ownerManager.setOwner(
+          ident.name(),
+          NameIdentifierUtil.toMetadataObject(ident, Entity.EntityType.MODEL),
+          PrincipalUtils.getCurrentUserName(),
+          Owner.Type.USER);
+    }
+    return model;
+  }
+
+  @Override
+  public boolean modelVersionExists(NameIdentifier ident, int version) {
+    return dispatcher.modelVersionExists(ident, version);
+  }
+
+  @Override
+  public boolean modelVersionExists(NameIdentifier ident, String alias) {
+    return dispatcher.modelVersionExists(ident, alias);
+  }
+
+  @Override
+  public Model alterModel(NameIdentifier ident, ModelChange... changes)
+      throws NoSuchModelException, IllegalArgumentException {
+    return dispatcher.alterModel(ident, changes);
+  }
+}
```

Comment:
Why don't we handle drop/alter model interface? I saw there's also the similar logic for drop/alter operations in fileset.

Reply:
Good question. Actually I hesitate to add similar logic. 
Because I am not sure whether the model needs the authz plugin. 
Can authentication of model be finished by Gravitino master. 
Because the locations under model may be a website. 
Maybe it doesn't have a common permission model system. 
If we have  S3 locations and website locations at the same time and we have authz plugin, we can push down the privileges of S3 but we can't push down the privileges of website. it maybe weird that we push down partial privileges. Is it better to let users guarantee the access of locations and don't add authz plugin.
So I think we don't need to add it now. It would be a better time when we need to implement some authz plugins of the model.

Reply:
At least, we should delete the owner when model is deleted, right?

Reply:
Yes, it should be implemented in `ModelMetaService` like `FilesetMetaService`. I can add them.

Reply:
We don't have to push the privileges to the underlying source, the purpose of model privilege and model owner is mainly for model metadata authentication.